port_ocean/utils.py

```diff
@@ -3,17 +3,40 @@
 from asyncio import ensure_future
 from functools import wraps
 from importlib.util import module_from_spec, spec_from_file_location
-from pathlib import Path
 from time import time
 from traceback import format_exception
 from types import ModuleType
 from typing import Callable, Any, Coroutine
 from uuid import uuid4
 
+import httpx
 import tomli
 import yaml
 from loguru import logger
+from pathlib import Path
 from starlette.concurrency import run_in_threadpool
+from werkzeug.local import LocalStack, LocalProxy
+
+from port_ocean.helpers.retry import RetryTransport
+
+_http_client: LocalStack[httpx.AsyncClient] = LocalStack()
+
+
+def _get_http_client_context() -> httpx.AsyncClient:
+    client = _http_client.top
+    if client is None:
+        client = httpx.AsyncClient(
+            transport=RetryTransport(
+                httpx.AsyncHTTPTransport(),
+                logger=logger,
+            )
+        )
+        _http_client.push(client)
+
+    return client
+
+
+http_async_client: httpx.AsyncClient = LocalProxy(lambda: _get_http_client_context())  # type: ignore
```

Comment:
Lets add comment describing that the purpose of this http_async_client is for integration use